include/wabt/compiler-support.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2022 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef WABT_COMPILER_SUPPORT_H_
+#define WABT_COMPILER_SUPPORT_H_
+
+// Taken from binaryen's src/compiler-support.h
+// The code might contain TODOs or stubs that read some values but do nothing
+// with them. The compiler might fail with [-Werror,-Wunused-variable].
+// The WASM_UNUSED(varible) is a wrapper that helps to suppress the error.
+#define WASM_UNUSED(expr)                                                      \
+  do {                                                                         \
+    if (sizeof expr) {                                                         \
+      (void)0;                                                                 \
+    }                                                                          \
+  } while (0)
```

Comment:
I think it would make sense to put this in `src/config.h.in` and call it `WABT_UNUSED`

Reply:
Oh wait.. we already have `WABT_UNUSED`.. i think maybe you can just apply that to definition of `success`?

Reply:
As a followup we should switch from `WABT_UNUSED` to C++17 `[[maybe_unused]]`?